### PR TITLE
feature: extends template definition

### LINKED ISSUE #377: feature request: Defining a template that extends another template
azure pipelines supports defining a template that extends another template.

for example:

`template-a.yml`:
```yml
parameters:
- name: paramA
  type: string
- name: paramB
  type: number

extends:
  template: `template-b.yml`:
  parameters:
...
```

`template-b.yml`:
```yml
parameters:
...
stages:
...
```

My issue here is that `public abstract class TemplateDefinition<T>` exposes a `public abstract ConditionedList<T> Definition { get; }` which will not work to render an `extends` object.

my current workarround is to define a type that extends the `ExtendsPipelineDefinition` type and set just the `Parameters` & `Extends` properties of the `PipelineWithExtends Pipeline` property

### PR DESCRIPTION
**Pull request by Meir017**
closes #377

### CODE REVIEW

File: src/Sharpliner/AzureDevOps/TemplateDefinition.cs [hunk 1]
```diff
@@ -220,7 +220,7 @@ public abstract class TemplateDefinition<T> : TemplateDefinition, ISharplinerDef
     /// Returns the definition of the template.
     /// </summary>
     [DisallowNull]
-    public abstract ConditionedList<T> Definition { get; }
+    public abstract T Definition { get; }
```
Comment (premun) [inline src/Sharpliner/AzureDevOps/TemplateDefinition.cs:223]: Why did this change?

Comment (Meir017) [reply src/Sharpliner/AzureDevOps/TemplateDefinition.cs:223]: I added `TemplateDefinitionBase<T>` which is the base class for `TemplateDefinition<ConditionedList<T>>`

this is in order to support templates of `extends` which contain two properties, `parameters` & `extends` but the `extends` is not a list, just a single object

Comment (premun) [reply src/Sharpliner/AzureDevOps/TemplateDefinition.cs:223]: I will need to have a closer look to understand if it should be `Conditioned<T>`

**Comment by Meir017** [comment]
I might have missed this, should we just extend `ExtendsPipelineDefinition<TPipeline>`? my main concern is that templates that extends other template can only declare the `parameters` & `extends` properties

**Comment by premun** [comment]
I will have a look

**Review by premun — APPROVED** [review]
Had a closer look and it looks good!

**Pull request merged**